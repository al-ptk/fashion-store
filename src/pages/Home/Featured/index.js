export { Featured as default } from './Featured';
