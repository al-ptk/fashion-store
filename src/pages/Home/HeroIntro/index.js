export { HeroIntro as default } from './HeroIntro';
