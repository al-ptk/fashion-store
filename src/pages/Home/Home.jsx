import { Link } from 'react-router-dom';

const Home = () => {
  return (
    <main aria-label="Home page">
      <section aria-labelledby="page-header">
        <h1 id="page-header">Brittania</h1>
      </section>

      <section aria-labelledby="featured-header">
        <h2 id="featured-header">Featured:</h2>
        {['012', '345', '678'].map((elem) => (
          <p>
            <Link key={elem} to={`/product/:${elem}`}>
              {elem}
            </Link>
          </p>
        ))}
      </section>

      <section aria-labelledby="catalog-header">
        <h2 id="catalog-header">Catalog:</h2>
        {['01', '23', '45', '67', '89'].map((elem) => (
          <p>
            <Link key={elem} to={`/catalog/?filter=${elem}`}>
              {elem}
            </Link>
          </p>
        ))}
      </section>
    </main>
  );
};

export default Home;
