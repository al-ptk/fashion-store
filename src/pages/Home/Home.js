import HeroIntro from './HeroIntro';

function Home() {
  return (
    <main>
      <HeroIntro />
    </main>
  );
}

export { Home };
