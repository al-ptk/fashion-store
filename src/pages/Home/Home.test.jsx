import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import Home from './Home';

describe('Home page', () => {
  it('renders', () => {
    render(<Home />);
    const main = screen.getByRole('main', { name: 'Home page' });
    expect(main).toBeInTheDocument();
  });

  describe('HeroIntro', () => {
    it('renders', () => {
      render(<Home />);
      const region = screen.getByRole('region', { name: 'Brittania' });
      expect(region).toBeInTheDocument();
    });
  });
});
