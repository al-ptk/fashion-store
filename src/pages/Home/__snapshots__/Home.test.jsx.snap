// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders properly 1`] = `
<main
  aria-label="Home page"
>
  <section
    aria-labelledby="page-heading"
  >
    <h1
      id="page-heading"
    >
      Brittania
    </h1>
  </section>
  <section
    aria-labelledby="featured-heading"
  >
    <h2
      id="featured-heading"
    >
      Featured:
    </h2>
    <a
      data-testid="featured-link"
      href="/product/012"
    >
      featured-012
    </a>
    <a
      data-testid="featured-link"
      href="/product/345"
    >
      featured-345
    </a>
    <a
      data-testid="featured-link"
      href="/product/678"
    >
      featured-678
    </a>
  </section>
  <section
    aria-labelledby="catalog-heading"
  >
    <h2
      id="catalog-heading"
    >
      Catalog:
    </h2>
    <a
      data-testid="catalog-link"
      href="/catalog/?filter=01"
    >
      catalog-01
    </a>
    <a
      data-testid="catalog-link"
      href="/catalog/?filter=23"
    >
      catalog-23
    </a>
    <a
      data-testid="catalog-link"
      href="/catalog/?filter=45"
    >
      catalog-45
    </a>
    <a
      data-testid="catalog-link"
      href="/catalog/?filter=67"
    >
      catalog-67
    </a>
    <a
      data-testid="catalog-link"
      href="/catalog/?filter=89"
    >
      catalog-89
    </a>
  </section>
  <footer>
    <section
      aria-labelledby="about-heading"
    >
      <h3
        id="about-heading"
      >
        About Us
      </h3>
      <a
        data-testid="about-link"
        href="/"
      >
        blog
      </a>
      <a
        data-testid="about-link"
        href="/"
      >
        newsletter
      </a>
      <a
        data-testid="about-link"
        href="/"
      >
        Institutional
      </a>
    </section>
    <section
      aria-labelledby="service-heading"
    >
      <h3
        id="service-heading"
      >
        Customer Service
      </h3>
      <a
        data-testid="service-link"
        href="/"
      >
        Our Email
      </a>
      <a
        data-testid="service-link"
        href="/"
      >
        Phone Number
      </a>
      <a
        data-testid="service-link"
        href="/"
      >
        Whatsapp
      </a>
    </section>
  </footer>
</main>
`;
